packages/graphql/src/schema/parse/get-unique-meta.ts

```diff
@@ -40,29 +34,14 @@ function getUniqueMeta(
 
     if (uniqueDirective) {
         const constraintName = uniqueDirective.arguments?.find((a) => a.name.value === "constraintName");
-        return {
-            constraintName: constraintName
-                ? (constraintName.value as StringValueNode).value
-                : `${type.name.value}_${fieldName}`,
-        };
+        if (constraintName && constraintName.value.kind === Kind.STRING) {
+            return { constraintName: constraintName.value.value };
+        }
+        return {};
     }
 
-    let uniqueId = false;
-
-    const idDirective = directives.find((x) => x.name.value === "id");
-
-    if (idDirective) {
-        const idDirectiveUniqueArgument = idDirective?.arguments?.find((a) => a.name.value === "unique")?.value as
-            | BooleanValueNode
-            | undefined;
-        // If unique argument is absent from @id directive, default is to use unique constraint
-        uniqueId = idDirectiveUniqueArgument ? idDirectiveUniqueArgument.value : true;
-    }
-
-    if (uniqueId) {
-        return {
-            constraintName: `${type.name.value}_${fieldName}`,
-        };
+    if (directives.some((d) => ["id", "relayId"].includes(d.name.value))) {
```

Comment:
Opinionated code style suggestion:

```suggestion
    if (directives.some((directive) => ["id", "relayId"].includes(directive.name.value))) {
```

Reply:
The strings `id`and `relayId` should really be constants somewhere.